flash/vision/data.py

```diff
@@ -0,0 +1,61 @@
+import pathlib
+from typing import Any, Iterable, Mapping, Optional, Sequence, Union
+
+from PIL import Image
+from torchvision.datasets.folder import default_loader, IMG_EXTENSIONS
+
+from flash.data.data_source import FilesDataSource, FoldersDataSource, SequenceDataSource
+
+
+class ImageFoldersDataSource(FoldersDataSource):
+
+    def __init__(
+        self,
+        train_folder: Optional[Union[str, pathlib.Path, list]] = None,
+        val_folder: Optional[Union[str, pathlib.Path, list]] = None,
+        test_folder: Optional[Union[str, pathlib.Path, list]] = None,
+        predict_folder: Optional[Union[str, pathlib.Path, list]] = None,
+    ):
+        super().__init__(
+            train_folder=train_folder,
+            val_folder=val_folder,
+            test_folder=test_folder,
+            predict_folder=predict_folder,
+            extensions=IMG_EXTENSIONS,
+        )
+
+    def load_sample(self, sample: Mapping[str, Any], dataset: Optional[Any] = None) -> Any:
+        result = {}  # TODO: this is required to avoid a memory leak, can we automate this?
```

Comment:
also both `load_sample` are the same

Reply:
Yeah, the issue is that each class needs to have different `load_data` methods but the same `load_sample`, I'm working on a way to clean this up haha